src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -440,9 +462,67 @@ class Manager : public DoutPrefixProvider {
         if (ret < 0) {
           ldpp_dout(this, 1) << "ERROR: failed to remove entries and/or lock queue up to: " << end_marker <<  " from queue: " 
             << queue_name << ". error: " << ret << dendl;
+          return;
         } else {
-          ldpp_dout(this, 20) << "INFO: removed entries up to: " << end_marker <<  " from queue: " 
-          << queue_name << dendl;
+          ldpp_dout(this, 20) << "INFO: removed entries up to: " << end_marker <<  " from queue: " << queue_name << dendl;
+        }
+
+        // reserving and committing the migrating entries
+        if (migration_in_process) {
+          std::vector<bufferlist> migration_vector;
+          uint64_t size_to_migrate = 0;
+          RGWPubSub ps(&rados_store, tenant_name);
+
+          rgw_pubsub_topic topic;
+          auto ret_of_get_topic = ps.get_topic(this, queue_name, topic, null_yield);
+          const auto& conf = this->get_cct()->_conf;
+
+          for (auto entry: entries_to_migrate) {
+            if (ret_of_get_topic < 0) {
+              // we can't migrate entries without topic info
+              return;
+            }
+            size_to_migrate += entry.data.length();
+            event_entry_t event_entry;
+            auto iter = entry.data.cbegin();
+            try {
+              decode(event_entry, iter);
+            } catch (buffer::error& err) {
+              ldpp_dout(this, 5) << "WARNING: failed to decode entry. error: " << err.what() << dendl;
+              break;
```

Comment:
if we failed to decode just one entry, maybe we should use `continue` and not `break`?

Reply:
Ack!